libdebug/snapshots/diff.py

```diff
@@ -0,0 +1,283 @@
+#
+# This file is part of libdebug Python library (https://github.com/libdebug/libdebug).
+# Copyright (c) 2024 Francesco Panebianco. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for details.
+#
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from libdebug.snapshots.memory.memory_map_diff import MemoryMapDiff
+from libdebug.snapshots.memory.memory_map_diff_list import MemoryMapDiffList
+from libdebug.snapshots.registers.register_diff import RegisterDiff
+from libdebug.snapshots.registers.register_diff_accessor import RegisterDiffAccessor
+from libdebug.utils.platform_utils import get_platform_register_size
+from libdebug.utils.pprint_primitives import (
+    pprint_diff_line,
+    pprint_diff_substring,
+    pprint_inline_diff,
+    pprint_memory_diff_util,
+    pprint_reg_diff_large_util,
+    pprint_reg_diff_util,
+)
+
+if TYPE_CHECKING:
+    from libdebug.snapshots.registers.snapshot_registers import SnapshotRegisters
+    from libdebug.snapshots.snapshot import Snapshot
+
+
+class Diff:
+    """This object represents a diff between two snapshots."""
+
+    def __init__(self: Diff, snapshot1: Snapshot, snapshot2: Snapshot) -> None:
+        """Initialize the Diff object with two snapshots.
+
+        Args:
+            snapshot1 (Snapshot): The first snapshot.
+            snapshot2 (Snapshot): The second snapshot.
+        """
+        if snapshot1.snapshot_id < snapshot2.snapshot_id:
+            self.snapshot1 = snapshot1
+            self.snapshot2 = snapshot2
+        else:
+            self.snapshot1 = snapshot2
+            self.snapshot2 = snapshot1
+
+        # The level of the diff is the lowest level among the two snapshots
+        if snapshot1.level == "base" or snapshot2.level == "base":
```

Comment:
You could probably do something like a StrEnum here, and just call min on it, but this is fine